<a href="https://colab.research.google.com/github/Sourav61/ML-AI-Prediciton-ShapeAi-/blob/main/Sourav_Pahwa_(2).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


Here i have imported some libraries:


In [ ]:
import pandas as pd
import numpy as np


Here we are exploring the iris data given by botanist using load_iris function imported from scikit-learn framework.

In [ ]:
from sklearn.datasets import load_iris
iris_dataset = load_iris()

Here we are printing key values of iris_dataset dictionary.iris_dataset was a dictionary returned by load_iris. 

In [ ]:
print("Keys of iris_dataset: \n{}".format(iris_dataset.keys()))

Keys of iris_dataset: 
dict_keys(['data', 'target', 'target_names', 'DESCR', 'feature_names', 'filename'])


Here we are printing first 200 charaters of description of the iris_dataset using slicing.

In [ ]:
val = iris_dataset['DESCR']
start_val = val[:200]
print( start_val + ":\n....")

.. _iris_dataset:

Iris plants dataset
--------------------

**Data Set Characteristics:**

    :Number of Instances: 150 (50 in each of three classes)
    :Number of Attributes: 4 numeric, predictive:
....


Here we are looking at value of the key "Target names" from dictionary iris_dataset.

In [ ]:
print("Target names: {}".format(iris_dataset['target_names']))

Target names: ['setosa' 'versicolor' 'virginica']


Here we are looking at value of the key "Feature names" from dictionary iris_dataset.

In [ ]:
print("Feature names : \n{}".format(iris_dataset['feature_names']))

Feature names : 
['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']


Here we are taking a look at the shape of the data using the shape method.

In [ ]:
print("Shape of data: {}".format(iris_dataset['data'].shape))

Shape of data: (150, 4)


Here we are looking the type of the data using the type function.

In [ ]:
print("Type of data: {}".format(type(iris_dataset['data'])))

Type of data: <class 'numpy.ndarray'>


Here we are printing first five rows of the data out of 150 datapoints and 4 features (petal length, sepal length, petal width, sepal width) using slicing operator.It is being printed in the form of 2-dimensional array having 5 rows and 4 columns.  

In [ ]:
print("First five rows of data :\n{}".format(iris_dataset['data'][:5]))

First five rows of data :
[[5.1 3.5 1.4 0.2]
 [4.9 3.  1.4 0.2]
 [4.7 3.2 1.3 0.2]
 [4.6 3.1 1.5 0.2]
 [5.  3.6 1.4 0.2]]


Here we are looking at the type of the target array.

In [ ]:
print("Type of target: {}".format(type(iris_dataset['target'])))

Type of target: <class 'numpy.ndarray'>


Here we are looking at the Shape of the target.

In [ ]:
print("Shape of target: {}".format(iris_dataset['target'].shape))

Shape of target: (150,)


Now here we are printing the entire target array.

In [ ]:
print("Target:\n{}".format(iris_dataset['target']))

Target:
[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 2 2 2 2 2 2 2 2 2 2 2
 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2
 2 2]


Here we are importing train_test_split from sklearn.model for training and testing our model(bot).

In [ ]:
from sklearn.model_selection import train_test_split
X_train , X_test, y_train, y_test = train_test_split(
    iris_dataset['data'], iris_dataset['target'], random_state=0)  

Here we arr looking at the dataset.

In [ ]:
print("X_train shape: {}".format(X_train.shape))
print("y_train shape: {}".format(y_train.shape))
print("X_test shape: {}".format(X_test.shape))
print("y_test shape: {}".format(y_test.shape))

X_train shape: (112, 4)
y_train shape: (112,)
X_test shape: (38, 4)
y_test shape: (38,)


Here we are importing a bot(model).

In [ ]:
from sklearn.neighbors import KNeighborsClassifier
knn = KNeighborsClassifier(n_neighbors=4)

Here we are fitting(training) our model.

In [ ]:
knn.fit(X_train, y_train)

KNeighborsClassifier(algorithm='auto', leaf_size=30, metric='minkowski',
                     metric_params=None, n_jobs=None, n_neighbors=1, p=2,
                     weights='uniform')

KNeighborsClassifier(algorithm='auto', leaf_size=30, metric='minkowski',
                     metric_params=None, n_jobs=None, n_neighbors=1, p=2,
                     weights='uniform')

Here we are looking at the array or learning of our model. 

In [ ]:
X_new = np.array([[5, 2.9, 1, 0.2]])

print("X_new.shape: {}".format(X_new.shape))

X_new.shape: (1, 4)


Her we are making prediction using our model.

In [ ]:
prediction = knn.predict(X_new)

print("Prediction: {}".format(prediction))
print("Predicted target name: {}".format(
iris_dataset['target_names'][prediction]))


Prediction: [0]
Predicted target name: ['setosa']


we are testing the model

In [ ]:
y_pred = knn.predict(X_test)
print("Test set predictions:\n {}".format(y_pred))

Test set predictions:
 [2 1 0 2 0 2 0 1 1 1 2 1 1 1 1 0 1 1 0 0 2 1 0 0 2 0 0 1 1 0 2 1 0 2 2 1 0
 2]


here are the results

In [ ]:
print("Test set score: {}".format(np.mean(y_pred == y_test)))

Test set score: 0.9736842105263158
